backend/controller/dal/pubsub.go

```diff
@@ -251,17 +263,30 @@ func (d *DAL) createSubscribers(ctx context.Context, tx *sql.Tx, key model.Deplo
 			if err != nil {
 				return fmt.Errorf("could not insert subscriber: %w", dalerrs.TranslatePGError(err))
 			}
+			logger.Debugf("Inserted subscriber %s for %s", subscriberKey, key)
 		}
 	}
 	return nil
 }
 
 func (d *DAL) removeSubscriptionsAndSubscribers(ctx context.Context, tx *sql.Tx, key model.DeploymentKey) error {
-	if err := tx.DeleteSubscriptions(ctx, key); err != nil {
+	logger := log.FromContext(ctx)
+
+	subscribers, err := tx.DeleteSubscribers(ctx, key)
+	if err != nil {
+		return fmt.Errorf("could not delete old subscribers: %w", dalerrs.TranslatePGError(err))
+	}
+	for _, s := range subscribers {
+		logger.Debugf("Deleted subscriber %s for %s", s, key)
+	}
+
+	subscriptions, err := tx.DeleteSubscriptions(ctx, key)
+	if err != nil {
 		return fmt.Errorf("could not delete old subscriptions: %w", dalerrs.TranslatePGError(err))
 	}
-	if err := tx.DeleteSubscribers(ctx, key); err != nil {
-		return fmt.Errorf("could not delete old subscribers: %w", dalerrs.TranslatePGError(err))
+	for _, s := range subscriptions {
+		logger.Debugf("Deleted subscription %s for %s", s, key)
```

Comment:
Should we maybe just log this once, but join the subscription names?
```suggestion
		logger.Debugf("Deleted subscriptions for %s", key, strings.Join(subscriptions, ", "))
```